model/src/main/java/com/epam/esm/repository/BaseRepository.java

```diff
@@ -0,0 +1,45 @@
+package com.epam.esm.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * The interface Base repository.
+ *
+ * @param <T> the type parameter
+ * @param <E> the type parameter
+ *
+ * @author Maksim Rutkouski
+ */
+public interface BaseRepository<T, E> {
+
```

Comment:
Seems that first generic value is always Long, so you can get rid of it

Reply:
I agree. In case of my project it's really not necessary. Should I correct it?